dhall/src/phase/resolve.rs

```diff
@@ -30,14 +31,14 @@ fn resolve_import(
     };
     match &import.location_hashed.location {
         Local(prefix, path) => {
-            let path: PathBuf = path.iter().cloned().collect();
-            let path = match prefix {
+            let path_buf: PathBuf = path.clone().into_iter().collect();
+            let path_buf = match prefix {
                 // TODO: fail gracefully
-                Parent => cwd.parent().unwrap().join(path),
-                Here => cwd.join(path),
+                Parent => cwd.parent().unwrap().join(path_buf),
+                Here => cwd.join(path_buf),
                 _ => unimplemented!("{:?}", import),
             };
-            Ok(load_import(&path, import_cache, import_stack).map_err(|e| {
+            Ok(load_import(&path_buf, import_cache, import_stack).map_err(|e| {
                 ImportError::Recursive(import.clone(), Box::new(e))
             })?)
         }
```

Comment:
Since this feature isn't tested by the tests from `dhall-lang`, could you add some tests here ? You don't need to test every case, but a few basic ones would ensure it works and we don't accidentally break it in the future.

Reply:
I think I just need to add the tests, and we're good to go here.

Reply:
Just to check I'm on the right page. I have a skeleton test where `?` is place holder for values:

```rust
    #[test]
    test_resolve_import_local() {
        let import = Import {
            mode: RawText,
            location: Local(?FilePrefix, FilePath { file_path: ?file_path }),
            hash: None,
        };
        let root = LocalDir(?cwd);
        let import_cache = HashMap::new();
        let import_stack = vec![];
        let import_result = resolve_import(import, root, import_cache, import_stack);
        assert_eq!(import_result, Ok(?some_result));
    }
```

Reply:
Oh I'm so sorry I missed this comment >< I was thinking more simply to just test `canonicalize`. For example
```
#[test]
test_canonicalize_whatever() {
    let filepath = FilePath {filepath = vec!["foo", "..", "bar"]};
    let expected = FilePath {filepath = vec!["bar"]};
    assert_eq!(filepath.canonicalize(), expected);
}
```

Reply:
Yeah sorry, that really wasn't clear from the diff this comment is pointing to